doc/content/reference/api/end_device.md

```diff
@@ -4,59 +4,114 @@ description: ""
+- `AsEndDeviceRegistry.Set` (Application Server)
+
+When deleting end devices, we recommend deleting them in the reverse order
+
 ## The `EndDeviceRegistry` service
 
-{{< proto/method service="EndDeviceRegistry" method="Create" >}}
+The Identity Server's `EndDeviceRegistry` is the first place to register an end device. This registry stores the following [EndDevice fields](#message:EndDevice):
 
-> TODO: [Document what fields to set to the IS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
+- `ids` (with subfields)
+- `name`
+- `description`
+- `attributes`
+- `version_ids` (with subfields)
+- `network_server_address`
+- `application_server_address`
+- `join_server_address` (only for OTAA devices)
+- `service_profile_id`
+- `locations`
+- `picture`
 
-{{< proto/method service="EndDeviceRegistry" method="Get" >}}
+{{< proto/method service="EndDeviceRegistry" method="Create" >}}
 
-> TODO: [Document what fields to get from the IS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
+{{< proto/method service="EndDeviceRegistry" method="Get" >}}
 
 {{< proto/method service="EndDeviceRegistry" method="List" >}}
 
-> TODO: [Document what fields to get from the IS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
-
 {{< proto/method service="EndDeviceRegistry" method="Update" >}}
 
-> TODO: [Document what fields to set to the IS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
-
 {{< proto/method service="EndDeviceRegistry" method="Delete" >}}
 
 ## The `JsEndDeviceRegistry` service
 
-{{< proto/method service="JsEndDeviceRegistry" method="Set" >}}
+OTAA devices are registered in the Join Server's `JsEndDeviceRegistry`. This registry stores the following [EndDevice fields](#message:EndDevice):
+
+- `ids` (with subfields)
+- `provisioner_id` (when provisioning with secure elements)
+- `provisioning_data` (when provisioning with secure elements)
+- `resets_join_nonces`
+- `root_keys`:
+  - `root_key_id`
+  - `app_key`
+  - `nwk_key`
+- `net_id`
+- `network_server_address`
+- `network_server_kek_label`
+- `application_server_address`
+- `application_server_id`
+- `application_server_kek_label`
+- `claim_authentication_code` (when using [end device claiming]({{< relref "end_device_claiming.md" >}}))
 
-> TODO: [Document what fields to set to the JS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
+{{< proto/method service="JsEndDeviceRegistry" method="Set" >}}
 
 {{< proto/method service="JsEndDeviceRegistry" method="Get" >}}
 
-> TODO: [Document what fields to get from the JS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
-
 {{< proto/method service="JsEndDeviceRegistry" method="Delete" >}}
 
 ## The `NsEndDeviceRegistry` service
 
-{{< proto/method service="NsEndDeviceRegistry" method="Set" >}}
+The Network Server's `NsEndDeviceRegistry` stores the following [EndDevice fields](#message:EndDevice):
+
+- `ids` (with subfields)
```

Comment:
Is this list meant to be exhaustive or is it more of an example? If it's the former - this list is not complete, e.g. `last_dev_status_received_at` (and more)

Reply:
Clarified with comment below